src/data_storage/sqlalchemy/data_source.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from functools import wraps
+
+from sqlalchemy.exc import DataError, NoResultFound
+from sqlalchemy.orm import sessionmaker
+
+from common.entities import schemas
+from data_storage.data_source import DataDriver
+from data_storage.sqlalchemy.models import Node, Option, Story, get_connection
+
+logging.basicConfig()
+logging.getLogger("sqlalchemy.engine").setLevel(logging.INFO)
+
+
+def execute_query(method):
+    @wraps(method)
+    def inner(self, *args, **kwargs):
+        with self.session.begin():
+            return method(self, *args, **kwargs)
+
+    return inner
+
+
+def handle_exception(method):
+    @wraps(method)
+    def inner(self, *args, **kwargs):
+        try:
+            data = method(self, *args, **kwargs)
+            return data
+        except (NoResultFound, DataError):
+            return None
+
+    return inner
+
+
+class RDSDriver(DataDriver):
+    def __init__(self, session=None):
+        if session is None:
+            engine = get_connection()
+            Sessionmaker = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+            self.session = Sessionmaker()
+        else:
+            self.session = session
+
+    @handle_exception
+    @execute_query
+    def get_story(self, story_id: str):
+        story = (
+            self.session.query(Story.id, Story.name, Node.name.label("root"))
+            .join(Node)
```

Comment:
Should be `isouter=True` ?

Reply:
Do you mean in case when `Root` `Node` is null?

Reply:
Yes, if we haven't had any nodes yet.